src/checkbox.scss

```diff
@@ -7,200 +7,136 @@
-    margin: $fd-checkbox-margin-compact;
-
-    &::after {
-      font-size: 0.625rem;
-    }
+  @mixin fd-checkbox-state($background-color, $text-color, $border-color, $border-size) {
+    + .#{$block}__label {
+      &::before {
+        background-color: $background-color;
+        border: $border-size solid $border-color;
+        color: $text-color;
+      }
 
-    &:indeterminate {
-      &::after {
-        width: 0.5rem;
-        height: 0.5rem;
+      @include fd-hover() {
+        @include fd-disabled() {
+          border-color: $border-color;
+        }
       }
     }
-
-    @include fd-form-radio-focus($fd-checkbox-margin-compact);
   }
 
-  // states
-  @include fd-hover() {
-    background-color: var(--sapField_Hover_Background);
-    border-color: var(--sapField_Hover_BorderColor);
+  $fd-checkbox-dimensions: 1.375rem;
+  $fd-checkbox-dimensions-compact: 1rem;
+  $fd-checkbox-margin: 0.6875rem;
+  $fd-checkbox-margin-compact: 0.5rem;
 
-    @include fd-disabled() {
-      border-color: var(--sapField_BorderColor);
-    }
-  }
+  @include fd-form-base();
 
-  &.is-invalid {
-    background-color: var(--sapField_InvalidBackground);
-    border: 0.125rem solid var(--sapField_InvalidColor);
+  @extend %fd-checkbox-input-hidden;
 
-    &:checked {
-      &::after {
-        color: var(--sapField_InvalidColor);
-      }
-    }
+  &__label {
+    @include fd-form-label();
+    @include action-cursor();
 
-    &:indeterminate {
-      &::after {
-        background-color: var(--sapField_InvalidColor);
-      }
+    display: flex;
+    align-items: center;
+
+    &::before {
+      content: "";
+      height: $fd-checkbox-dimensions;
+      width: $fd-checkbox-dimensions;
+      font-size: var(--sapFontLargeSize);
+      display: inline-flex;
+      justify-content: center;
+      align-items: center;
+      font-family: "SAP-icons";
+      background-color: var(--sapField_Background);
+      margin: $fd-checkbox-margin;
+      color: var(--sapSelectedColor);
+      border: var(--sapField_BorderWidth) solid var(--sapField_BorderColor);
+      border-radius: var(--sapField_BorderCornerRadius);
     }
 
     @include fd-hover() {
-      @include fd-disabled() {
-        border-color: var(--sapField_InvalidColor);
+      &::before {
+        background-color: var(--sapField_Hover_Background);
+        border-color: var(--sapField_Hover_BorderColor);
       }
     }
   }
 
-  &.is-valid {
-    background-color: var(--sapField_SuccessBackground);
-    border: var(--sapField_BorderWidth) solid var(--sapField_SuccessColor);
-
-    &:checked {
-      &::after {
-        color: var(--sapField_SuccessColor);
-      }
-    }
+  &:checked + .#{$block}__label::before {
+    content: "\e05b";
+  }
 
-    &:indeterminate {
-      &::after {
-        background-color: var(--sapField_SuccessColor);
-      }
-    }
+  &:indeterminate + .#{$block}__label::before {
+    padding-bottom: 0.0625rem;
+    content: "\e17b";
+    font-size: 0.875rem;
```

Comment:
It should be 0.75 x 0.75 rem sized square

Reply:
It should be and it is. In general `font-size: 0.75rem` doesn't look good. It looks same as before, when we add 0.75rem + 0.0625rem (1px)